class2.1/task3.cpp

```diff
@@ -0,0 +1,74 @@
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <string>
+//I create history.txt file for saving data.
+using namespace std;
+
+map<string, int> userHistory;
+
+void loadHistory() {
```

Comment:
Using global object inside function is not a good idea cause it will be difficult to test such a function.
The same for other functions. But what if we push this object out of the function => it will become an argument and we will have something like this
```
void loadHistory(auto& userHistory);
void saveHistory(const auto& userHistory);
void greetUser(auto& userHistory, const auto& name);
void resetStatistics(auto& userHistory, const auto& name);
void clearHistory(auto& userHistory);
```
But it looks just like C++ implementation of member methods of class - each method takes this as the first argument under the hood => We should encapsulate the data into a class UserHistory or database.
Why it is important? Firstt of all you will hide the data layout and structures from client code and you will provide a public API for manipulations while at any moment you will be able to change that std::map to any other structure if needed, you will be able to extend the data or saving format without any impact on client code.
Another important benefit is that it will be easy to write unit test for that class and it's methods especially if you will pass history file as a parameter of constructor.

Also note that your class should handle only data as it is the main responsibility. Different logs and interpretations should be managed by clients of the class. For example increment counter for user may return new counter as a return value and then client code can analyze it and print output.